lib/SILGen/SILGenDestructor.cpp

```diff
@@ -499,6 +499,8 @@ void SILGenFunction::emitMoveOnlyMemberDestruction(SILValue selfValue,
                                                    NominalTypeDecl *nom,
                                                    CleanupLocation cleanupLoc,
                                                    SILBasicBlock *finishBB) {
+  // drop_deinit must be used to invalidate any user-defined struct/enum deinit
+  // before the individual members can be destroyed.
   selfValue = B.createDropDeinit(cleanupLoc, selfValue);
   if (selfValue->getType().isAddress()) {
```

Comment:
One thing that I don't understand is why here and in the drop_deinit lowering in the ownership model eliminator we are doing destructuring? Seems like it should be in one place, no?

Reply:
Ownership lowering needs to correctly lower the destroy_value operation. There's no way around that.

It's always valid to destructure an aggregate that has no deinit. But it shouldn't be done unless needed, so I don't understand why SILGen was doing it. It was only hiding downstream bugs. At this point, I don't think it makes sense to preserve any of the existing behavior, so I went ahead and removed that useless destructure from SILGen.